modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java

```diff
@@ -141,14 +151,36 @@ public static void recoverDistributedSlotMap() throws AndesException {
      */
     private static void recoverMapsForEachQueue() throws AndesException {
         List<AndesQueue> queueList = contextStore.getAllQueuesStored();
-
-        for (AndesQueue queue : queueList) {
+        List<Future> futureSlotRecoveryExecutorList = new ArrayList<Future>();
+        ExecutorService executorService = Executors.newFixedThreadPool(5);
```

Comment:
There could be multiple storage queues worked before entire cluster went down. We need to recover all remaining message of each storage queue at the first node start up in the cluster. So this is done parallely but Cassandra shouldn't be exhausted because there will be multiple database read call at the same time by different threads. 5 is just a optimal number I decided. I will make it to configurable number so user can increase it based on number of storage queues exist to speed up warm start up.